src/pop/blockchain/pop/vbk_block_tree.cpp

```diff
@@ -211,7 +213,10 @@ bool VbkBlockTree::validateBTCContext(const VbkBlockTree::payloads_t& vtb,
                                return height <= vtb.containingBlock.getHeight();
                              });
 
-  return isValid ? true : state.Invalid("block-referenced-too-early");
+  return isValid
+             ? (invalid_vtbs.erase(vtb.getId()), true)
```

Comment:
Is it possible that missing BTC block will be provided with another VTB? Those bad VTBs will stay forever in such a case.
You can try to store missing BTC blocks in a flat structure, eg std::vector and check if provided BTC context fixes the gap.